File: Java/src/com/leetcode/Easy/SumOfDigitsOfStringAfterConvert.java
package com.leetcode.Easy;

// 1945. Sum of Digits of String After Convert
public class SumOfDigitsOfStringAfterConvert {
    public int getLucky(String s, int k) {
        StringBuilder sb = new StringBuilder();
        for (char ch : s.toCharArray()) {
            sb.append(ch - 'a' + 1);
        }

        while (k > 0) {
            int sum = 0;
            for (int i = 0; i < sb.length(); i++) {
                sum += Integer.parseInt(String.valueOf(sb.charAt(i)));
            }
            k--;
            sb.setLength(0);
            sb.append(sum);
        }

        return Integer.parseInt(sb.toString());
    }
}

File: JavaScript/Easy/SumOfDigitsOfStringAfterConvert.js
/**
 * 1945. Sum of Digits of String After Convert
 * @param {string} s
 * @param {number} k
 * @return {number}
 */
const getLucky = function (s, k) {
  let numericString = '';
  for (let i = 0; i < s.length; i++) {
    numericString += (s.charCodeAt(i) - 96);
  }

  while (k > 0) {
    let sum = 0;
    for (let i = 0; i < numericString.length; i++) {
      sum += parseInt(numericString[i], 10);
    }
    numericString = sum.toString();
    k--;
  }

  return parseInt(numericString, 10);
};